src/rgw/rgw_file.h

```diff
@@ -698,13 +700,25 @@ namespace rgw {
     }
 
     void decode(bufferlist::const_iterator& bl) {
-      DECODE_START(2, bl);
+      DECODE_START(3, bl);
       uint32_t fh_type;
       decode(fh_type, bl);
       if ((fh.fh_type != fh_type) &&
 	 (fh_type == RGW_FS_TYPE_SYMBOLIC_LINK))
         fh.fh_type = RGW_FS_TYPE_SYMBOLIC_LINK;  
       ceph_assert(fh.fh_type == fh_type);
+      if (struct_v >= 3) {
+	int64_t fov;
+	decode(fov, bl);
+	if (fh.fh_type == RGW_FS_TYPE_FILE) {
+	  if (file_ondisk_version >= fov) {
+	    // the current version is newer than the ondisk version
+	    return;
+	  }
+	  // make sure the following "encode" always encode a greater version
+	  file_ondisk_version = fov + 1;
```

Comment:
So it appears that the only place that decode is called is from RGWFileHandle::decode_attrs. So I think this would be simplified if that higher-level logic were moved here.